### DISCUSSION TITLE
S3 exporter: endpoint documentation / bucket

### DISCUSSION THREAD

**Discussion by dabcoder** [open-telemetry/opentelemetry-collector-contrib#32766] [General]
[The S3 exporter documentation](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/awss3exporter/README.md#exporter-configuration) states that the `endpoint` parameter overrides the endpoint used by the exporter instead of constructing it from `region` and `s3_bucket` . 

Yet [in the code](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e0af2457aed5ba6107da21204eb41d84273ca8dc/exporter/awss3exporter/config.go#L54) there is:

```
if c.S3Uploader.S3Bucket == "" {
  errs = multierr.Append(errs, errors.New("bucket is required"))
}
```

However, when the `endpoint` is provided, it should also contain the S3 bucket, thus I would not expect the `s3_bucket` to have to be provided and this error to be returned. Am I misunderstanding the config setup here?

**Comment by dmitryax** [comment]
Seems like a bug in validation. @dabcoder feel free to submit a PR. cc @atoulme @pdelewski as code owners

**Reply by dmitryax** [reply]
Converted into an issue https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32774

**Reply by dabcoder** [reply]
Thanks for the prompt response. Yes I will see if I can tackle it and will let you know how it goes.

**Comment by dabcoder** [comment]
Closing this as this can be tracked in the above issue (and linked PR)
